kolibri/core/utils/portal.py

```diff
@@ -0,0 +1,63 @@
+import json
+import socket
+
+import requests
+from morango.api.serializers import CertificateSerializer
+from morango.certificates import Certificate
+from morango.certificates import Nonce
+from morango.constants.api_urls import NONCE
+from six.moves.urllib.parse import urljoin
+
+from kolibri.core.auth.constants.morango_scope_definitions import FULL_FACILITY
+from kolibri.utils import conf
+
+
+def claim(token, facility, client_ip=None):
+
+    # get client ip if not passed in
+    if not client_ip:
+        hostname = socket.gethostname()
+        client_ip = socket.gethostbyname(hostname)
```

Comment:
I feel like I've hit situations where these can throw exceptions, but I can't remember. Also, do we really need the IP here? I feel like it's mostly used with Nonces on the server end, to make sure the client using the nonce is the one that created it. Could we allow `None` or put in a dummy value for this case?

Reply:
Discussed: remove client nonce entirely (it's just one-way proving for this command)